app/server.py

```diff
@@ -19,108 +20,269 @@
+	H is for headers, X is used to specify HTTP Method, d is used to pass a message.
+	"""
+
+	data=request.get_json()
+	data['id'] = item_id
+
+	if is_valid(data, 'item'):
+		try:
+			wl = Wishlist.find_or_404(wishlist_id)
+			wl.update_item(data)
+			wl.save_wishlist()
+			new_wl = wl.find(wishlist_id)
+			return make_response(jsonify(new_wl.serialize_wishlist()), status.HTTP_200_OK)
+		except WishlistException:
+			message = { 'error' : 'Wishlist %s was not found' % wishlist_id }
+			return make_response(jsonify(message), status.HTTP_404_NOT_FOUND)
+		except ItemException:
+			message = { 'error' : 'Item %s was not found' % item_id }
+			return make_response(jsonify(message), status.HTTP_404_NOT_FOUND)
+	else:
+		message = {'error' : 'Item data was not valid'}
+		return make_response(jsonify(message), status.HTTP_400_BAD_REQUEST)
+
+
+@app.route('/wishlists/<int:wishlist_id>/items/<string:item_id>', methods=['DELETE'])
+def remove_wishlist_item(wishlist_id, item_id):
+	"""
+	The route for removing a specific item in a wishlist,
+	given a wishlist_id and the item_id
+	Example: curl -X DELETE http://127.0.0.1:5000/wishlists/1/items/i123
+	"""
+
+	try:
+		wl = Wishlist.find_or_404(wishlist_id)
+		wl.remove_item(item_id)
+		wl.save_wishlist()
+		return make_response('', status.HTTP_204_NO_CONTENT)
+	except ItemException:
+		return make_response(jsonify(message='Item with id %s could not be found' % item_id), status.HTTP_204_NO_CONTENT)
+	except WishlistException:
+		return make_response(jsonify(message='Wishlist with id %d could not be found' % wishlist_id), status.HTTP_204_NO_CONTENT)
+
+
+
+@app.route('/wishlists/<int:wishlist_id>/items/clear', methods=['PUT'])
+def clear_wishlist(wishlist_id):
+	"""
+		The route for clearing a wishlist specified by wishlist_id
+		without deleting the wishlist itself.
+		Example: curl -X PUT http://127.0.0.1:5000/wishlists/1/items/clear
+	"""
+
+	try:
+		wl = Wishlist.find_or_404(wishlist_id)
+		wl.remove_item(None)
+		wl.save_wishlist()
+		new_wl = wl.find(wishlist_id)
+		return make_response(jsonify(new_wl.serialize_wishlist()), status.HTTP_200_OK)
+	except WishlistException:
+		message = { 'error' : 'Wishlist %s was not found' % wishlist_id }
+		return make_response(jsonify(message), status.HTTP_404_NOT_FOUND)
+
+
+@app.route('/wishlists/<int:wishlist_id>', methods=['DELETE'])
+def delete_wishlist(wishlist_id):
+	"""
+	The route for deleting a specific wishlist when the wishlist_id is specified.
+	This only does a soft delete, i.e. update the deleted flag with "true"
+	Example: curl -X DELETE http://127.0.0.1:5000/wishlists/1
+	"""
+
+	try:
+		wl = Wishlist.find_or_404(wishlist_id)
+		wl.delete()
+		return make_response('', status.HTTP_204_NO_CONTENT)
+	except WishlistException:
+		return make_response(jsonify(message='Wishlist with id %d could not be found' % wishlist_id), status.HTTP_204_NO_CONTENT)
+
+
+@app.route('/wishlists/search', methods=['GET'])
+def search_wishlists():
```

Comment:
How is the url of search actuall? I try both 'curl http://0.0.0.0.:5000/wishlists/search?q=new_item' and 'curl http://0.0.0.0.:5000/wishlists/search#q=new_item', or 'curl http://0.0.0.0.:5000/wishlists/search?user_id=123'. It just return no wishlist.